Fakebook.Profile/Fakebook.Profile.DataAccess/EntityModel/EntityProfile.cs

```diff
@@ -1,50 +1,123 @@
 {
     /// <summary>
     /// Database model of a profile.
     /// </summary>
+    [Table("Profile")]
     public class EntityProfile
     {
+        #nullable enable
+
+        /// <summary>
+        /// Create a profile with default profile picture
+        /// </summary>
+        /// <param name="email">The user's email. Must be unique.</param>
+        /// <param name="firstName">The user's firstname.</param>
+        /// <param name="lastName">The user's lastname.</param>
+        /// <param name="dob">The user's date of birth.</param>
+        /// <param name="phoneNumber">The user's phone number.</param>
+        /// <param name="status">The user's status.</param>
+        /// <param name="profilePictureUrl">The url of the user's profile picture.</param>
+        public EntityProfile(string email, string firstName, string lastName, DateTime dob, string? phoneNumber = null, string? status = null, IList<string>? followerEmails = null, IList<string>? followingEmails = null)
+        {
+            Email = email;
+            FirstName = firstName;
+            LastName = lastName;
+            BirthDate = dob;
+            PhoneNumber = phoneNumber;
+            FollowerEmails = followerEmails;
+            FollowingEmails = followingEmails;
+            ProfilePictureUrl = new Uri(ProfileConfiguration.DefaultUri);
+        }
+
         /// <summary>
-        /// The primary key, the user's email.
+        /// Create a profile with user-specified profile picture
+        /// </summary>
+        /// <param name="email">The user's email. Must be unique.</param>
+        /// <param name="firstName">The user's firstname.</param>
+        /// <param name="lastName">The user's lastname.</param>
+        /// <param name="dob">The user's date of birth.</param>
+        /// <param name="phoneNumber">The user's phone number.</param>
+        /// <param name="status">The user's status.</param>
+        /// <param name="profilePictureUrl">The url of the user's profile picture.</param>
+        public EntityProfile(string email, string firstName, string lastName, DateTime dob, Uri profilePictureUrl, string? phoneNumber = null, string? status = null, IList<string>? followerEmails = null, IList<string>? followingEmails = null)
+        {
+            Email = email;
+            FirstName = firstName;
+            LastName = lastName;
+            BirthDate = dob;
+            PhoneNumber = phoneNumber;
+            FollowerEmails = followerEmails;
+            FollowingEmails = followingEmails;
+            ProfilePictureUrl = profilePictureUrl;
+        }
+
+        /// <summary>
+        /// The user's ID.  The primary key.  Cannot be null.
         /// </summary>
         [Key]
-        [Column(name: nameof(Email))]
+        [Required]
+        public int UserId { get; set; }
+
+        /// <summary>
+        /// The user's unique email. Cannot be null.
+        /// </summary>
+        [Required]
+        [MaxLength(45),MinLength(5)]
```

Comment:
there's actually a huge amount of variety possible with emails. in practice it's usually best to accept anything with a @ in it, so take off these length constraints